Makefile.build

```diff
@@ -121,10 +121,9 @@ image-nomos: license
 		--build-arg VERSION=${VERSION} \
 		.
 
-
 # Pushes the specified component's docker image to gcr.io.
-push-to-gcr-nomos: image-nomos
```

Comment:
I'm curious about the change. Why did we remove the dependency?
The original intent is what we push is what we build.

After the change, if we run `make image-nomos` first and followed by `make push-to-gcr-nomos`, the image URL will be different with different timestamp. We have to manually provide `GCR_PREFIX` with the timestamp of `make image-nomos`, which is not obvious to get.

Reply:
@sdowell ^^